static/js/Sheets/SheetsWithRefPage.jsx

```diff
@@ -0,0 +1,191 @@
+                sheets = sheets.sort((a, b) => new Date(b.dateCreated) - new Date(a.dateCreated));
+            break;
+        }
+        return sheets;
+    }
+    const prepSheetsForDisplay = (sheets) => {
+        sheets = sheets.sort((a, b) => {
+                              // First place user's sheet
+                              if (a.owner === Sefaria.uid && b.owner !== Sefaria.uid) {
+                                return -1;
+                              }
+                              if (a.owner !== Sefaria.uid && b.owner === Sefaria.uid) {
+                                return 1;
+                              }
+                              // Then sort by language / interface language
+                              let aHe, bHe;
+                              [aHe, bHe] = [a.title, b.title].map(Sefaria.hebrew.isHebrew);
+                              if (aHe !== bHe) { return (bHe ? -1 : 1) * (Sefaria.interfaceLang === "hebrew" ? -1 : 1); }
+                            })
+        return sheets;
+    }
+    const normalizeSheetsMetaData = (sheets) => {
+        return sheets.map(sheet => {
+            return {
+                        sheetId: sheet.id,
+                        title: sheet.title,
+                        owner_name: sheet.ownerName,
+                        owner_image: sheet.ownerImageUrl,
+                        profile_url: sheet.ownerProfileUrl,
+                        dateCreated: sheet.dateCreated,
+                        _id: sheet.id,
+                        snippet: sheet.summary || "",
+                    }
+        })
+    }
+    const handleSheetsLoad = (sheets) => {
+      searchState.availableFilters = Sefaria.sheets.sheetsWithRefFilterNodes(sheets);
+      setSheets(sheets);
+      updateSearchState(searchState, 'sheet');
+      setOrigAvailableFilters(searchState.availableFilters);
+      setLoading(false);
+    }
+    const getSheetsByRefCallback = (sheets) => {
+        // filters out duplicate sheets by sheet ID number and filters so that we don't show sheets as connections to themselves
+        return sheets.filter((sheet, index, self) =>
+                          index === self.findIndex((s) => (
+                            s.id === sheet.id)
+                        ))
+    }
+    const updateOrigAvailableFilters = () => {
+        // update component prop 'origAvailableFilters' based on changes to availableFilters in 'searchState'
+        origAvailableFilters.forEach(availableFilter => {
+            const selected = searchState.appliedFilters.includes(availableFilter.aggKey);
+            if (selected && selected !== Boolean(availableFilter.selected)) {
+                availableFilter.setSelected(true);
+            } else if (selected !== Boolean(availableFilter.selected)) {
+                availableFilter.setUnselected(true);
+            }
+        })
+    }
+
+    useEffect(() => {
+      // 'collections' won't be present if the related API set _sheetsByRef,
```

Comment:
This breaks encapsulation with the cache. Instead store results for this page using a separate key (see my comments on `Sefaria.getSheetsByRef()`. This useEffect should not access the cache at all.